exporter/awskinesisexporter/internal/batch/batch_test.go

```diff
@@ -15,13 +15,24 @@
 package batch_test
 
 import (
+	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awskinesisexporter/internal/batch"
 )
 
+func TestBatchRecordsCapacity(t *testing.T) {
+	t.Parallel()
+
+	b := batch.New()
+	assert.NotNil(t, b, "Must not be nil")
+
+	recordsCap := reflect.ValueOf(b).Elem().FieldByName("records").Cap()
```

Comment:
I am not a fan of using reflection, ideally we should use the values directly. 

Considering that this file is using the test package namespace, I suggest making a new file a such:

```golang
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

package batch

func TestBatchRecordsCapacity(t *testing.T) {
  t.Parallel()

  b := New()
  
  assert.Equal(t, MaxBatchedRecords, cap(b.records), "Must have the correct value set")
}
```